pkg/controller/acicontainersoperator.go

```diff
@@ -311,124 +389,134 @@ func (t *OperatorHandler) CheckOwnerReference(reference []metav1.OwnerReference)
-	t.HostDaemonset, _ = hostdaemonsetclient.Get("aci-containers-host",metav1.GetOptions{})
-	if t.HostDaemonset == nil {
+	c.Resources.HostDaemonset, _ = hostdaemonsetclient.Get("aci-containers-host", metav1.GetOptions{})
+	if c.Resources.HostDaemonset == nil {
 		log.Info("aci-containers-host daemonset is nil.....returning")
-		return
+		return false
 	}
 
-	if  !t.CheckOwnerReference(t.HostDaemonset.OwnerReferences){
-		t.HostDaemonset.OwnerReferences = []metav1.OwnerReference{
+	if !c.CheckOwnerReference(c.Resources.HostDaemonset.OwnerReferences) {
+		c.Resources.HostDaemonset.OwnerReferences = []metav1.OwnerReference{
 			*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
 		}
 
-		_, err = hostdaemonsetclient.Update(t.HostDaemonset)
+		_, err := hostdaemonsetclient.Update(c.Resources.HostDaemonset)
 		if err != nil {
 			log.Error(err.Error())
+			return false
 		}
 	}
+	log.Info("Successfully updated owner reference to the aci-container-host daemonset")
+	return true
+
+}
 
-	ovsdaemonsetclient := k8sclient.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE"))
+func (c *Controller) UpdateOvsDaemonsetOwnerReference(acicontainersoperator *operators.AciContainersOperator) bool {
+	ovsdaemonsetclient := c.K8s_Clientset.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE"))
 	if ovsdaemonsetclient == nil{
 		log.Info("Error in Fetching ovsdaemonsetclient...")
-		return
+		return false
 	}
 
-	t.OvsDaemonset, _ = hostdaemonsetclient.Get("aci-containers-openvswitch",metav1.GetOptions{})
-	if t.OvsDaemonset == nil {
+	c.Resources.OvsDaemonset, _ = ovsdaemonsetclient.Get("aci-containers-openvswitch",metav1.GetOptions{})
+	if c.Resources.OvsDaemonset == nil {
 		log.Info("aci-containers-openvswitch daemonset is nil.....returning")
-		return
+		return false
 	}
 
-	if  !t.CheckOwnerReference(t.OvsDaemonset.OwnerReferences){
-		t.OvsDaemonset.OwnerReferences = []metav1.OwnerReference{
+	if  !c.CheckOwnerReference(c.Resources.OvsDaemonset.OwnerReferences){
+		c.Resources.OvsDaemonset.OwnerReferences = []metav1.OwnerReference{
 			*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
 		}
 
-		_, err = ovsdaemonsetclient.Update(t.OvsDaemonset)
+		_, err := ovsdaemonsetclient.Update(c.Resources.OvsDaemonset)
 		if err != nil {
 			log.Error(err.Error())
+			return false
 		}
 	}
+	log.Info("Successfully updated owner reference to the aci-containers-openvswitch daemonset")
+	return true
+}
+
+func (c *Controller) handleOperatorCreate(obj interface{}) bool {
+
+	log.Info("OperatorHandler.ObjectCreated")
+
+	acicontainersoperator := obj.(*operators.AciContainersOperator)
+
+	log.Debug(acicontainersoperator.Spec.Config)
+
+	if (acicontainersoperator.Spec.Config == ""){
+		log.Info("acicnioperator CR config is Nil")
+		return true
+	}
+	
+	dec, err := base64.StdEncoding.DecodeString(acicontainersoperator.Spec.Config)
+	if err != nil {
+		panic(err)
```

Comment:
if you want to retry , i guess you need to remove panic. applicable every where you have put panic

Reply:
Noted .. handling it